Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,493 @@
+                    // It will typically be provided in the form of a variable. We do not support other scenarios at this time.
+                    if (!(invokeCommandAst.CommandElements[i + 1] is VariableExpressionAst variableAst))
+                    {
+                        break;
+                    }
+
+                    sessionName = variableAst.VariablePath.UserPath;
+                    return true;
+                }
+
+                sessionName = null;
+                return false;
+            }
+
+            /// <summary>
+            /// GetAssignedVarsInSession: Retrieves all previously declared vars for a given session (as in Invoke-Command -Session $session).
+            /// </summary>
+            /// <param name="sessionName"></param>
+            private IEnumerable<VariableExpressionAst> GetAssignedVarsInSession(string sessionName)
+            {
+                return _varsDeclaredPerSession[sessionName];
+            }
+
+            /// <summary>
+            /// AddAssignedVarsToSession: Adds variables to the list of assigned variables for a given Invoke-Command session.
+            /// </summary>
+            /// <param name="sessionName"></param>
+            /// <param name="variablesToAdd"></param>
+            private void AddAssignedVarsToSession(string sessionName, IEnumerable<VariableExpressionAst> variablesToAdd)
+            {
+                if (!_varsDeclaredPerSession.ContainsKey(sessionName))
+                {
+                    _varsDeclaredPerSession.Add(sessionName, new List<VariableExpressionAst>());
+                }
+
+                _varsDeclaredPerSession[sessionName].AddRange(variablesToAdd);
+            }
+
+            /// <summary>
+            /// AnalyzeScriptBlock: Generate a Diagnostic Record for each incorrectly used variable inside a given ScriptBlock.
+            /// </summary>
+            /// <param name="scriptBlockExpressionAst"></param>
+            private void AnalyzeScriptBlock(ScriptBlockExpressionAst scriptBlockExpressionAst)
+            {
+                    IEnumerable<VariableExpressionAst> nonAssignedNonUsingVarAsts = FindNonAssignedNonUsingVarAsts(scriptBlockExpressionAst,
+                    FindVarsInAssignmentAsts(scriptBlockExpressionAst));
+
+                GenerateDiagnosticRecords(nonAssignedNonUsingVarAsts);
+            }
+
+            /// <summary>
+            /// GenerateDiagnosticRecords: Add Diagnostic Records to the internal list for each given variable
+            /// </summary>
+            /// <param name="nonAssignedNonUsingVarAsts"></param>
+            private void GenerateDiagnosticRecords(IEnumerable<VariableExpressionAst> nonAssignedNonUsingVarAsts)
+            {
+                foreach (var variableExpression in nonAssignedNonUsingVarAsts)
+                {
+                    if (variableExpression == null)

```

Comment:
This check shouldn't be needed if the caller is performing as expected. A null check makes sense with a public API, but with a private one it will just disguise a possible bug